specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -3,7 +3,7 @@
   "info": {
     "title": "ComputeDiagnosticResourceProviderClient",
     "description": "The Compute Diagnostic Resource Provider Client",
-    "version": "2024-03-01-preview"
+    "version": "2021-06-01-preview"
```

Comment:
why change to 2021?

Reply:
Hi Zhenhua, this is because the new 2024-03-01-preview version is blocked by the ARM compute manifest rollout taking very long to deploy. https://msazure.visualstudio.com/One/_release?definitionId=18535&view=mine&_a=releases 

Without the ARM manifest rolled out, the APIs in this swagger are unable to invoke using 2024-03-01-preview API version. This is blocking our Az PS PR from merging in by the current **release milestone cutoff 3/25**. If possible, please let us know if we can merge this ASAP to unblock the Az PS PR. Thank you 

https://github.com/Azure/azure-powershell/pull/24352

Reply:
AFAIK, it's not allowed. The API version in the file should be consistent with the name of the folder which includes the file, and the content in the readme. Since `2021-06-01-preview` already exists, it will cause two files point to the same API version. And since you do not update the readme, the related SDK may not be generated.